Contributor policy updates - Code of Conduct and AI guidelines (#1830)
htop-dev/htop: CONTRIBUTING.md

Author: Agreed, I think moving style guide earlier will be helpful, I'll add a followup-commit.

The details of the coding style as well as what to take care about with your
contributions can be found in our [style guide](docs/styleguide.md).

Use of AI
---------

You are welcome to use whatever tools work best for you, which includes use
of AI tools to assist with developing htop. AI tools have proven invaluable
as learning aids for new contributors, helped to identify problematic code,
and have been used to fill cross-platform knowledge gaps.

With great power comes great responsibility. Please ensure you review and
understand any changes made by tools you use before opening pull requests.

> Be aware that AI tools may generate content which is an exact copy of code

Reviewer: I prefer to remove the them "exact":

```suggestion
Be aware that AI tools may generate code or content that is copied
```

From the legal perspective it does not have to be exact copy of code to constitute copyright infringement. If the AI generates a substantial, unattributed block of code, that's enough to raise suspicions on the copyright.

My personal attitude towards AI (not suggesting as an htop policy) is: don't use it for generating anything more than trivial code. Trivial patches may be okay for AI because it would be easy to argue (e.g. the "merger doctrine" as a legal defense). But IANAL (I am not a lawyer). Don't take that as a serious legal advice.
